### PR TITLE
Final attempt

### PR DESCRIPTION
**Pull request by s-ieremenko**

### CODE REVIEW

**Comment by CatDidenko** [comment]
Remove app.js file from the repository.

File: app.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
+import { createServer, IncomingMessage, ServerResponse } from 'http';
+import * as fs from 'fs';
+
+const host: string = '127.0.0.1';
+const port: number = 3000;
+const path: string = './newFile.txt';
```
Comment (CatDidenko) [inline app.ts]: Move constants to the separate file for constants.

File: app.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+createServer((request: IncomingMessage, response: ServerResponse) => {
+    switch (request.method) {
+        case 'GET': {
+            fs.readFile(path, 'utf-8', (error, data) => {
```
Comment (CatDidenko) [inline app.ts]: Move encoding to the constants file.
Wrong formatting. Use 2 spaces instead of 4.

File: app.ts [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+                if (error) {
+                    response.writeHead(404, { 'Content-Type': 'text/plain' })
+                    response.end('Resourse not found!');
+                }
+                else {
+                    response.writeHead(200, { 'Content-Type': 'text/plain' })
+                    response.end(data);
+                }
```
Comment (CatDidenko) [inline app.ts]: You can avoid "else" if you add return in error block
Снимок экрана 2021-07-08 в 09 15 51

File: app.ts [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+                    })
+                }
+            })
+            break;
+        }
+
+        default: {
+            response.writeHead(405, { 'Content-Type': 'text/plain' })
+            response.end('Method Not Allowed');
+            break
```
Comment (CatDidenko) [inline app.ts]: Add semicolon;

**Review by CatDidenko — CHANGES_REQUESTED** [review]
Please, add package.json file to the repo

File: app.ts [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+                    request.on('data', chunk => {
+                        body += chunk;
+                    })
+                    request.on('end', () => {
+                        fs.appendFile(path, body += '\n', err => {
+                            if (err) {
+                                response.writeHead(500, { 'Content-Type': 'text/plain' })
+                                response.end('Internal Server Error');
+                            } else {
+                                response.writeHead(200, { 'Content-Type': 'text/plain' })
```
Comment (CatDidenko) [inline app.ts]: You can move duplicated headers to the constant

**Review by CatDidenko — CHANGES_REQUESTED** [review]

File: app.ts (hunk 1 above)
Comment (s-ieremenko) [reply app.ts]: I've made all the corrections, tell me please if it's correct to write the explanation in one commit or I should add each separate commit to an appropriate file. Thank you

**Review by CatDidenko — APPROVED** [review]

**Pull request merged**